zerver/lib/user_groups.py

```diff
@@ -387,8 +387,10 @@ def user_groups_in_realm_serialized(realm: Realm) -> list[UserGroupDict]:
         "can_mention_group__named_user_group",
     ).filter(realm=realm)
 
-    membership = UserGroupMembership.objects.filter(user_group__realm=realm).values_list(
-        "user_group_id", "user_profile_id"
+    membership = (
+        UserGroupMembership.objects.filter(user_group__realm=realm)
+        .exclude(user_profile__is_active=False, user_group__named_user_group__isnull=False)
```

Comment:
What's the story with the `user_group__named_user_group__isnull=` part here?

I'm finding it a bit hard to wrap my head around whether the exclude should be two separate excludes or a single one.

Reply:
Yeah, I wanted to ask this. This is for including the deactivated members in setting groups, i.e. anonymous groups. Ideally we would want to exclude deactivated users from members list for those as well, but that would need some more changes as well like to send the events to update the settings as well, so just wanted to confirm.